File: frontend/src/types/session.ts
export interface SessionInfo {
  id: string;
  preview: string | null;
  updated_at: string;
}

export interface HistoryMessage {
  role: 'user' | 'assistant';
  content: string;
}

export interface HistoryData {
  messages: HistoryMessage[];
  summary: string | null;
}

File: backend/schemas/session.py
from pydantic import BaseModel, Field
from typing import Optional


class SessionInfo(BaseModel):
    id: str
    preview: Optional[str] = None
    updated_at: str = ""


class SessionListResponse(BaseModel):
    sessions: list[SessionInfo] = Field(default_factory=list)


class SessionCreateResponse(BaseModel):
    session_id: str


class HistoryMessage(BaseModel):
    role: str
    content: str


class HistoryResponse(BaseModel):
    messages: list[HistoryMessage] = Field(default_factory=list)
    summary: Optional[str] = None
